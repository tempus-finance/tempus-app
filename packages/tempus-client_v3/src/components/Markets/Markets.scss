.tc__app__markets {
  display: flex;
  flex-direction: column;
  flex-grow: 1;
  justify-content: space-between;
}
