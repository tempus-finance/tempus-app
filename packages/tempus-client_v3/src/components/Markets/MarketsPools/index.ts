import MarketsPools from './MarketsPools';

export default MarketsPools;
