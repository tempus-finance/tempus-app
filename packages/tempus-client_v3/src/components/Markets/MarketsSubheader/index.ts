import type { FilterType, ViewType, PoolType, SortType, SortOrder } from './MarketsSubheader';
import MarketsSubheader from './MarketsSubheader';

export { FilterType, ViewType, PoolType, SortType, SortOrder };
export default MarketsSubheader;
