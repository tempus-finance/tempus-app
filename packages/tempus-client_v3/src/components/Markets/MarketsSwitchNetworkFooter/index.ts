import MarketsSwitchNetworkFooter from './MarketsSwitchNetworkFooter';

export default MarketsSwitchNetworkFooter;
