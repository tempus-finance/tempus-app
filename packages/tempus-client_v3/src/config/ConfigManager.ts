import { Chain, ChainConfig, Config, ProtocolName, TempusPool, Ticker } from 'tempus-core-services';
import config from './config';

interface TokenListItem {
  chain: Chain;
  address: string;
}

class ConfigManager {
  private config: Config = config;

  private chainList: Chain[] = [];
  private poolList: TempusPool[] = [];
  private tokenList: TokenListItem[] = [];

  init(): void {
    this.retrieveChainList();
    this.retrievePoolList();
    this.retrieveTokenList();
  }

  getConfig(): Config {
    return this.config;
  }

  getChainConfig(chain: Chain): ChainConfig {
    return this.config[chain];
  }

  getChainList(): Chain[] {
    return this.chainList;
  }

  getPoolList(): TempusPool[] {
    return this.poolList;
  }

  getTokenList(): TokenListItem[] {
    return this.tokenList;
  }

  // TODO add tests
  getFilteredPoolList(filterByChain?: Chain, filterByToken?: Ticker, filterByProtocol?: ProtocolName): TempusPool[] {
    return this.getPoolList()
      .filter(pool => {
        if (filterByChain) {
          return pool.chain === filterByChain;
        }

        return true;
      })
      .filter(pool => {
        if (filterByToken) {
          return pool.backingToken === filterByToken;
        }

        return true;
      })
      .filter(pool => {
        if (filterByProtocol) {
          return pool.protocol === filterByProtocol;
        }

        return true;
      });
  }

  getEarliestStartDate(filterByChain?: Chain, filterByToken?: Ticker, filterByProtocol?: ProtocolName): Date {
    const earliestPoolList = this.getPoolList()
      .filter(pool => {
        if (filterByChain) {
          return pool.chain === filterByChain;
        }

        return true;
      })
      .filter(pool => {
        if (filterByToken) {
          return pool.backingToken === filterByToken;
        }

        return true;
      })
      .filter(pool => {
        if (filterByProtocol) {
          return pool.protocol === filterByProtocol;
        }

        return true;
      })
      .sort((poolA, poolB) => (poolA.startDate < poolB.startDate ? -1 : 1));

    if (earliestPoolList && earliestPoolList.length) {
      return new Date(earliestPoolList[0].startDate);
    }

    throw new Error('getEarliestStartDate - Cannot find a pool by search criteria');
  }

  getMaturityDates(filterByChain?: Chain, filterByToken?: Ticker, filterByProtocol?: ProtocolName): Date[] {
    const today = new Date();
    const todayTime = today.getTime();

    const filteredPools = this.getFilteredPoolList(filterByChain, filterByToken, filterByProtocol)
      .filter(pool => todayTime <= pool.maturityDate)
      .sort((poolA, poolB) => (poolA.maturityDate < poolB.maturityDate ? -1 : 1));

    const dates: Date[] = [];
    filteredPools.forEach(pool => dates.push(new Date(pool.maturityDate)));

    return dates;
  }

  private retrieveChainList(): void {
    this.chainList = Object.keys(this.config) as Chain[];
  }

  private retrievePoolList(): void {
    const pools: TempusPool[] = [];

    Object.keys(this.config).forEach((chain: string) => {
      this.getChainConfig(chain as Chain).tempusPools.forEach((pool: TempusPool) => {
        pools.push({ ...pool, chain: chain as Chain });
      });
    });

    this.poolList = pools;
  }

  private retrieveTokenList(): void {
    const result: TokenListItem[] = [];

    const poolList = this.getPoolList();
    poolList.forEach(pool => {
      const poolChain = pool.chain;

      if (poolChain) {
        result.push(
          {
            address: pool.backingTokenAddress,
            chain: poolChain,
          },
          {
            address: pool.yieldBearingTokenAddress,
            chain: poolChain,
          },
          {
            address: pool.principalsAddress,
            chain: poolChain,
          },
          {
            address: pool.yieldsAddress,
            chain: poolChain,
          },
          {
            address: pool.ammAddress,
            chain: poolChain,
          },
        );
      } else {
        console.warn(`Pool ${pool.address} does not contain chain in it's config!`);
      }
    });

    this.tokenList = result;
  }
}

export default ConfigManager;
