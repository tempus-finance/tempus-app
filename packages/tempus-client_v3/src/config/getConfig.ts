import { Octokit } from 'octokit';
import { Config } from 'tempus-core-services';

const TempusPoolsConfig = {
  owner: 'tempus-finance',
  repo: 'tempus-pools-config',
  path: 'config.json',
};

let config: Config;
export const getConfig = async (): Promise<Config> => {
  if (!config) {
    try {
      const octokit = new Octokit();

      const response = await octokit.rest.repos.getContent({
        mediaType: {
          format: 'json',
        },
        ...TempusPoolsConfig,
      });

      if (response.status === 200) {
        const { data } = response;
        const { content } = data as any;

        if (content) {
          const decodedString = await String(Buffer.from(content, 'base64'));
          config = JSON.parse(decodedString);
        }
      }
    } catch (error) {
      console.log('Error retrieving Tempus Pools Config', error);
      return {};
    }
  }

  return config;
};
