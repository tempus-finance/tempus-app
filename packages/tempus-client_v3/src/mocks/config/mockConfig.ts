import { ChainConfig, TempusPool } from 'tempus-core-services';

export const pool1: TempusPool = {
  address: '1',
  poolId: 'ethereum-1',
  backingToken: 'ETH',
  backingTokenAddress: '0x0000000000000000000000000000000000000000',
  yieldBearingToken: 'stETH',
  yieldBearingTokenAddress: '00001-ybt',
  principalsAddress: '00001-p',
  yieldsAddress: '00001-y',
  ammAddress: '00001-amm',
  chain: 'ethereum',
  protocol: 'lido',
  protocolDisplayName: 'Lido',
  spotPrice: '1',
  decimalsForUI: 4,
  startDate: 1640995200000, // 2022/01/01
  maturityDate: 1735689600000, // 2025/01/01
  tokenPrecision: {
    backingToken: 18,
    yieldBearingToken: 18,
    lpTokens: 18,
    principals: 18,
    yields: 18,
  },
  disabledOperations: {},
  showEstimatesInBackingToken: false,
  rippleColor: '#ffffaa',
};

export const pool2: TempusPool = {
  address: '2',
  poolId: 'ethereum-2',
  backingToken: 'USDC',
  backingTokenAddress: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48',
  yieldBearingToken: 'yvUSDC',
  yieldBearingTokenAddress: '00002-ybt',
  principalsAddress: '00002-p',
  yieldsAddress: '00002-y',
  ammAddress: '00002-amm',
  chain: 'ethereum',
  protocol: 'yearn',
  protocolDisplayName: 'Yearn',
  spotPrice: '1',
  decimalsForUI: 4,
  startDate: 1643673600000, // 2022/02/01
  maturityDate: 1709251200000, // 2024/03/01
  tokenPrecision: {
    backingToken: 6,
    yieldBearingToken: 6,
    lpTokens: 18,
    principals: 6,
    yields: 6,
  },
  disabledOperations: {},
  showEstimatesInBackingToken: true,
  rippleColor: '#fafafa',
};

export const pool3: TempusPool = {
  address: '3',
  poolId: 'fantom-1',
  backingToken: 'USDC',
  backingTokenAddress: '0x04068da6c83afcfa0e13ba15a6696662335d5b75',
  yieldBearingToken: 'yvUSDC',
  yieldBearingTokenAddress: '00003-ybt',
  principalsAddress: '00003-p',
  yieldsAddress: '00003-y',
  ammAddress: '00003-amm',
  chain: 'fantom',
  protocol: 'yearn',
  protocolDisplayName: 'Yearn',
  spotPrice: '1',
  decimalsForUI: 2,
  startDate: 1642204800000, // 2022/15/01
  maturityDate: 1743724800000, // 2025/04/04
  tokenPrecision: {
    backingToken: 6,
    yieldBearingToken: 6,
    lpTokens: 18,
    principals: 6,
    yields: 6,
  },
  disabledOperations: {},
  showEstimatesInBackingToken: true,
  rippleColor: '#fcfcfc',
};

export const pool4: TempusPool = {
  address: '4',
  poolId: 'fantom-2',
  backingToken: 'WETH',
  backingTokenAddress: '0x74b23882a30290451A17c44f4F05243b6b58C76d',
  yieldBearingToken: 'yvWETH',
  yieldBearingTokenAddress: '00004-ybt',
  principalsAddress: '00004-p',
  yieldsAddress: '00004-y',
  ammAddress: '00004-amm',
  chain: 'fantom',
  protocol: 'yearn',
  protocolDisplayName: 'Yearn',
  spotPrice: '1',
  decimalsForUI: 4,
  startDate: 1647216000000, // 2022/03/14
  maturityDate: 1671062400000, // 2022/12/15
  tokenPrecision: {
    backingToken: 18,
    yieldBearingToken: 18,
    lpTokens: 18,
    principals: 18,
    yields: 18,
  },
  disabledOperations: {},
  showEstimatesInBackingToken: true,
  rippleColor: '#122112',
};

export const pool5: TempusPool = {
  address: '5',
  poolId: 'fantom-3',
  backingToken: 'WETH',
  backingTokenAddress: '0x74b23882a30290451A17c44f4F05243b6b58C76d',
  yieldBearingToken: 'yvWETH',
  yieldBearingTokenAddress: '00004-ybt',
  principalsAddress: '00004-p',
  yieldsAddress: '00004-y',
  ammAddress: '00004-amm',
  chain: 'fantom',
  protocol: 'yearn',
  protocolDisplayName: 'Yearn',
  spotPrice: '1',
  decimalsForUI: 4,
  startDate: 1647216000000, // 2022/03/14
  maturityDate: 1676419200000, // 2023/2/15
  tokenPrecision: {
    backingToken: 18,
    yieldBearingToken: 18,
    lpTokens: 18,
    principals: 18,
    yields: 18,
  },
  disabledOperations: {},
  showEstimatesInBackingToken: true,
  rippleColor: '#ababab',
};

export const mockConfig: { [chain: string]: ChainConfig } = {
  ethereum: {
    tempusPools: [pool1, pool2],
    chainId: 1,
    privateNetworkUrl: String(process.env.REACT_APP_ETHEREUM_RPC),
    alchemyKey: String(process.env.REACT_APP_MAINNET_ALCHEMY_KEY),
    averageBlockTime: 13.3,
    blockExplorerName: 'Etherscan',
    blockExplorerUrl: 'https://etherscan.io/',
    lidoOracle: '',
    nativeToken: 'ETH',
    nativeTokenPrecision: 18,
    networkName: 'ethereum',
    publicNetworkUrl: '',
    statisticsContract: '',
    tempusControllerContract: '',
    vaultContract: '',
  },
  fantom: {
    tempusPools: [pool3, pool4, pool5],
    chainId: 250,
    privateNetworkUrl: String(process.env.REACT_APP_FANTOM_RPC),
    averageBlockTime: 0.9,
    blockExplorerName: 'FTMScan',
    blockExplorerUrl: 'https://ftmscan.com/',
    lidoOracle: '',
    nativeToken: 'FTM',
    nativeTokenPrecision: 18,
    networkName: 'fantom',
    publicNetworkUrl: '',
    statisticsContract: '',
    tempusControllerContract: '',
    vaultContract: '',
    alchemyKey: '',
  },
  'ethereum-fork': {
    tempusPools: [],
    chainId: 31337,
    privateNetworkUrl: String(process.env.REACT_APP_ETHEREUM_RPC),
    alchemyKey: String(process.env.REACT_APP_MAINNET_ALCHEMY_KEY),
    averageBlockTime: 13.3,
    blockExplorerName: 'Etherscan',
    blockExplorerUrl: 'https://etherscan.io/',
    lidoOracle: '',
    nativeToken: 'ETH',
    nativeTokenPrecision: 18,
    networkName: 'ethereum',
    publicNetworkUrl: '',
    statisticsContract: '',
    tempusControllerContract: '',
    vaultContract: '',
  },
};

jest.mock('../../config/config', () => ({
  __esModule: true,
  default: mockConfig,
}));
