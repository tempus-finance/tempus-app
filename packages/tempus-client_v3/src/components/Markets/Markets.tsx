import { FC } from 'react';
import { Outlet } from 'react-router-dom';
import MarketsSubheader from './MarketsSubheader';
import MarketsPools from './MarketsPools';
import MarketsSwitchNetworkFooter from './MarketsSwitchNetworkFooter';
import './Markets.scss';

const Markets: FC = () => (
  <div className="tc__app__markets">
    <div className="tc__app__markets-content">
      <MarketsSubheader />
      <MarketsPools />
    </div>
    <MarketsSwitchNetworkFooter />
    <Outlet />
  </div>
);

export default Markets;
