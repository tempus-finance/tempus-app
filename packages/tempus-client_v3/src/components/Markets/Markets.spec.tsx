import { fireEvent, render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { of as mockOf } from 'rxjs';
import { Decimal, Decimal as MockDecimal } from 'tempus-core-services';
import { getConfigManager } from '../../config/getConfigManager';
import {
  useActivePoolList,
  useInactivePoolList,
  useMaturedPoolList,
  useTvlData,
  usePoolBalances,
  useFixedAprs,
} from '../../hooks';
import Markets from './Markets';

const subject = () =>
  render(
    <BrowserRouter>
      <Markets />
    </BrowserRouter>,
  );

jest.mock('../../hooks', () => ({
  ...jest.requireActual('../../hooks'),
  useActivePoolList: jest.fn(),
  useInactivePoolList: jest.fn(),
  useMaturedPoolList: jest.fn(),
  usePoolBalances: jest.fn(),
  useTvlData: jest.fn(),
}));

jest.mock('../../hooks/useFixedAprs', () => ({
  ...jest.requireActual('../../hooks/useFixedAprs'),
  useFixedAprs: jest.fn(),
  poolAprs$: mockOf({
    'ethereum-1': new MockDecimal(0.041),
    'ethereum-2': new MockDecimal(0.038),
    'fantom-3': new MockDecimal(0.18),
    'fantom-4': new MockDecimal(0.106),
    'fantom-5': new MockDecimal(0.126),
  }),
}));

describe('Markets', () => {
  beforeAll(async () => {
    const config = getConfigManager();
    config.init();
  });

  beforeEach(() => {
    (useActivePoolList as jest.Mock).mockImplementation(() => getConfigManager().getPoolList().slice(0, -2));
    (useInactivePoolList as jest.Mock).mockImplementation(() => getConfigManager().getPoolList().slice(-2, -1));
    (useMaturedPoolList as jest.Mock).mockImplementation(() => getConfigManager().getPoolList().slice(-1));
    (usePoolBalances as jest.Mock).mockReturnValue({
      'ethereum-1': new Decimal(500),
      'ethereum-2': new Decimal(700),
      'fantom-3': new Decimal(200),
      'fantom-4': new Decimal(900),
      'fantom-5': new Decimal(800),
    });
    (useTvlData as jest.Mock).mockReturnValue({
      'ethereum-1': new Decimal(5000),
      'ethereum-2': new Decimal(7000),
      'fantom-3': new Decimal(2000),
      'fantom-4': new Decimal(9000),
      'fantom-5': new Decimal(8000),
    });
    (useFixedAprs as jest.Mock).mockReturnValue({
      'ethereum-1': new Decimal(0.041),
      'ethereum-2': new Decimal(0.038),
      'fantom-3': new Decimal(0.18),
      'fantom-4': new Decimal(0.106),
      'fantom-5': new Decimal(0.126),
    });
  });

  it('renders the page', () => {
    const { container } = subject();

    expect(container).not.toBeNull();
    expect(container).toMatchSnapshot();
  });

  it('render pools after updates pool type to boosted', () => {
    const { container } = subject();

    const poolTypeButtons = container.querySelectorAll('.tc__tabs__tab');

    fireEvent.click(poolTypeButtons[1]);

    expect(container).toMatchSnapshot();
  });

  it('render pools after updates filters', () => {
    const { container, getByRole } = subject();

    const filterButton = getByRole('button', { name: 'Filter' });
    expect(filterButton).not.toBeNull();

    fireEvent.click(filterButton);

    const filterTypeRadios = container.querySelectorAll('.tc__dropdown:first-of-type input[type=radio]');
    expect(filterTypeRadios).toHaveLength(3);

    filterTypeRadios.forEach(filterTypeRadio => {
      fireEvent.click(filterTypeRadio);

      expect(container).toMatchSnapshot();
    });
  });

  it('render pools after updates sorting types', () => {
    const { container, getByRole } = subject();

    const sortButton = getByRole('button', { name: 'Sort' });
    expect(sortButton).not.toBeNull();

    fireEvent.click(sortButton);

    const sortTypeButtons = container.querySelectorAll('.tc__dropdown:last-of-type .tc__dropdownItem button');
    expect(sortTypeButtons).toHaveLength(5);

    sortTypeButtons.forEach(button => {
      fireEvent.click(button);

      expect(container).toMatchSnapshot();

      fireEvent.click(button);

      expect(container).toMatchSnapshot();
    });
  });
});
