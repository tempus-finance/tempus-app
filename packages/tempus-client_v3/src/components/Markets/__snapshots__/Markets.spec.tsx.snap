// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markets renders the page 1`] = `
<div>
  <div
    class="tc__app__markets"
  >
    <div
      class="tc__nav-subheader"
    >
      <div
        class="tc__nav-subheader__background"
      />
      <div
        class="tc__nav-subheader__wrapper"
      >
        <div
          class="tc__nav-subheader__group"
        >
          <div
            class="tc__tabs tc__tabs__tabs-small"
          >
            <button
              class="tc__btn tc__tabs__tab"
              data-selected="false"
              type="button"
            >
              <div
                style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
              >
                Fixed
              </div>
            </button>
            <button
              class="tc__btn tc__tabs__tab"
              data-selected="false"
              type="button"
            >
              <div
                style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
              >
                Boosted
              </div>
            </button>
            <button
              class="tc__btn tc__tabs__tab"
              data-selected="true"
              type="button"
            >
              <div
                style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(255, 255, 255); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
              >
                All
              </div>
            </button>
            <div
              class="tc__tabs__indicator-container"
            >
              <div
                class="tc__tabs__indicator"
                style="width: 33.333333333333336%; left: 66.66666666666667%;"
              />
            </div>
          </div>
        </div>
        <div
          class="tc__nav-subheader__group"
        >
          <div
            class="tc__dropdown"
          >
            <button
              class="tc__btn tc__tooltip-wrapper-anchor"
              data-selected="false"
              type="button"
            >
              <div
                style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
              >
                Filter
              </div>
              <svg
                class="tc__icon tc__icon-down-chevron"
                fill="none"
                height="12"
                viewBox="0 0 16 16"
                width="12"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  clip-path="url(#clip0_909_5439)"
                >
                  <path
                    d="M15.7387 5.36992L8.63082 11.7636C8.54823 11.8385 8.44996 11.898 8.34169 11.9385C8.23342 11.9791 8.11729 12 8 12C7.88271 12 7.76658 11.9791 7.65831 11.9385C7.55004 11.898 7.45177 11.8385 7.36918 11.7636L0.261296 5.36992C0.0939908 5.21942 -2.49304e-09 5.01531 0 4.80248C2.49304e-09 4.58965 0.0939908 4.38554 0.261296 4.23504C0.428601 4.08455 0.655515 4 0.89212 4C1.12873 4 1.35564 4.08455 1.52294 4.23504L8 10.0693L14.4771 4.23504C14.6444 4.08455 14.8713 4 15.1079 4C15.3445 4 15.5714 4.08455 15.7387 4.23504C15.906 4.38554 16 4.58965 16 4.80248C16 5.01531 15.906 5.21942 15.7387 5.36992Z"
                    fill="#222222"
                  />
                </g>
                <defs>
                  <clippath
                    id="clip0_909_5439"
                  >
                    <rect
                      fill="white"
                      height="16"
                      width="16"
                    />
                  </clippath>
                </defs>
              </svg>
            </button>
          </div>
          <div
            class="tc__dropdown"
          >
            <button
              class="tc__btn tc__tooltip-wrapper-anchor"
              data-selected="false"
              type="button"
            >
              <div
                style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
              >
                Sort
              </div>
              <svg
                class="tc__icon tc__icon-down-chevron"
                fill="none"
                height="12"
                viewBox="0 0 16 16"
                width="12"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  clip-path="url(#clip0_909_5439)"
                >
                  <path
                    d="M15.7387 5.36992L8.63082 11.7636C8.54823 11.8385 8.44996 11.898 8.34169 11.9385C8.23342 11.9791 8.11729 12 8 12C7.88271 12 7.76658 11.9791 7.65831 11.9385C7.55004 11.898 7.45177 11.8385 7.36918 11.7636L0.261296 5.36992C0.0939908 5.21942 -2.49304e-09 5.01531 0 4.80248C2.49304e-09 4.58965 0.0939908 4.38554 0.261296 4.23504C0.428601 4.08455 0.655515 4 0.89212 4C1.12873 4 1.35564 4.08455 1.52294 4.23504L8 10.0693L14.4771 4.23504C14.6444 4.08455 14.8713 4 15.1079 4C15.3445 4 15.5714 4.08455 15.7387 4.23504C15.906 4.38554 16 4.58965 16 4.80248C16 5.01531 15.906 5.21942 15.7387 5.36992Z"
                    fill="#222222"
                  />
                </g>
                <defs>
                  <clippath
                    id="clip0_909_5439"
                  >
                    <rect
                      fill="white"
                      height="16"
                      width="16"
                    />
                  </clippath>
                </defs>
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tc__marketsPools-container"
    />
  </div>
</div>
`;
